dashboard/legacy/middleware/files_api.rb

```diff
@@ -421,13 +415,17 @@ def put_file(endpoint, encrypted_channel_id, filename, body, project_type = nil)
       quota_crossed_half_used(endpoint, encrypted_channel_id) if quota_crossed_half_used?(app_size, body.length)
     end
 
-    # Block libraries with PII/profanity from being published.
-    # Block main.json file from aichat lab flagged with profanity from being saved.
-    #
-    # The "backpack" feature uses libraries to allow students to share code
-    # between their own projects -- skip this check for .java and .py files, since in this use case
-    # the files are only being used by a single user.
-    if endpoint == 'libraries' && BACKPACK_PROGRAM_FILE_TYPES.exclude?(file_type)
+    unless project_type
+      project = Projects.new(get_storage_id).get(encrypted_channel_id)
+      project_type = project[:projectType]&.downcase if project
```

Comment:
Is there ever a case where an app lab project would not have the project type set? Trying to figure out if the check below should also include a `nil` project type for profanity filtering.

Reply:
In `projects.rb` in the `create method`, `project_type` is assigned `type || level&.project_type` . So if the request is from ‘projects/applab/new’ or 'projects/applab/<channel-id>/remix, then `project_type` is assigned `params[:key]`, i.e., ‘applab’ for an App Lab standalone project.
And if an App Lab level is created, then `project_type` is assigned from the `project_type` of the `Level` which is `game&.app`. For App Lab this is also `applab’. So I think that we don't need to include a `nil` project type below. Lmk if this sounds correct to you. Thanks!

Reply:
@molly-moen I updated condition to skip if project_type is 'backpack' so  re-requesting your review. I was having issues with the unit test on drone so thought this was also a valid way to implement.

Reply:
nice!